### PR TITLE
binomial exponential power

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #51

### CODE REVIEW

**Comment by Graur** [comment]
@maxonfjvipon  Please, restart your build

**Comment by Graur** [comment]
@maxonfjvipon You can increase stack size in your pom.xml. Or wait until [this PR](https://github.com/objectionary/eo/pull/1405) will be released.

**Comment by maxonfjvipon** [comment]
@Graur ok, I'll wait

**Comment by ivan-egorov42** [comment]
@Graur I can add this to pom.xml here as I suppose this won't cover third-party libraries: https://github.com/objectionary/eo/issues/1424

**Comment by Graur** [comment]
@ivan-egorov42 I think I've already solve this issue by refactoring. Also I've already create PR with stack size increasing% #76.

**Comment by maxonfjvipon** [comment]
@Graur please have a look

File: src/main/eo/org/eolang/math/number.eo [hunk 1]
```diff
@@ -192,72 +192,144 @@
...
-            powf
-              neg.
-                x-as-float
-      *
-        x-as-float.gt 0.0
-        if.
-          self-as-float.eq 0.0
+      []
+        if. > x-as-float!
+          x-as-num.is-infinite > x-is-inf
```
Comment (Graur) [inline src/main/eo/org/eolang/math/number.eo:204]: @maxonfjvipon Can `x-is-inf` be constant ?

File: src/main/eo/org/eolang/math/number.eo [hunk 2]
```diff
@@ -192,72 +192,144 @@
...
-    and. > both-are-int!
-      (number x).is-int
-      is-int
-
-    if. > x-as-float!
-      (number x).is-nan > x-is-nan!
+    if. > @
+      or.
+        is-nan.
+          number x > x-as-num
```
Comment (Graur) [inline src/main/eo/org/eolang/math/number.eo:199]: Can `x-as-num` be constant ?

**Review by Graur — CHANGES_REQUESTED** [review]
@maxonfjvipon Good one, just a few comments from my side

File: src/main/eo/org/eolang/math/number.eo (hunk 1 above)
Comment (maxonfjvipon) [reply src/main/eo/org/eolang/math/number.eo:204]: @Graur sure

**Comment by maxonfjvipon** [comment]
@Graur please have a look one more time

**Review by Graur — APPROVED** [review]
@maxonfjvipon Great, thanks!
@yegor256 Please, take a look

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**